test/TestAssets/MultitargetedNetFrameworkProject/UnitTest1.cs

```diff
@@ -68,18 +71,22 @@ public void FailsUntilNet462ButPassesOnNewerNetFramework()
                 exception = ex;
             }
 
-            switch (TargetFramework)
+            var shouldThrowException = new[] { "NET451", "NET452", "NET46", "NET461", "NET462" }.Contains(TargetFramework);
+
+            if (shouldThrowException && exception == null)
+            {
+                throw new Exception($"Expected the code above to throw exception, " +
+                    $"because it fails when running in <= NET462 versions of .NET Framework, " +
+                    $"and we are running in process that was compiled as {TargetFramework}, " +
+                    $"and is called {processName}, but there was no exception.");
+            }
+
+            if (!shouldThrowException && exception != null)
```

Comment:
If you merge this condition with the previous you can avoid duplicating the same exception.

Reply:
But the exception is different, that is why there are 2 ifs. :)

Reply:
The first one was for positive case, and the other one for negative.